pkg/apis/numaflow/v1alpha1/vertex_types.go

```diff
@@ -228,7 +226,18 @@ func (v Vertex) GetPodSpec(req GetVertexPodSpecReq) (*corev1.PodSpec, error) {
 	envVars := []corev1.EnvVar{
 		{Name: EnvVertexObject, Value: encodedVertexSpec},
 	}
-	envVars = append(envVars, v.commonEnvs()...)
+
+	commonEnvVars := v.commonEnvs()
+	for _, vtx := range req.PipelineSpec.Vertices {
+		if vtx.IsASource() && vtx.Source.Serving != nil {
+			commonEnvVars = append(commonEnvVars,
+				corev1.EnvVar{Name: EnvCallbackEnabled, Value: "true"},
```

Comment:
It seems like we changed the logic from getting the information from annotations.

Are we expecting it's always `true`?

Reply:
This PR  removes the need for having an annotation to enable callbacks. Instead, if a serving source is used, the environment variable to enable callbacks will be set automatically. The `EnvCallbackEnabled` is checked to construct the callback http client, the callback only happens if the message contains the `X-Numaflow-Callback-Url` header.